generator/codegen/linkfollower_extensions.go

```diff
@@ -0,0 +1,392 @@
+package codegen
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"go/ast"
+	"go/printer"
+	"go/token"
+	"path/filepath"
+	"strings"
+
+	_ "github.com/ARM-software/embedded-development-services-client/client"
+	"github.com/ARM-software/golang-utils/utils/commonerrors"
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/getkin/kin-openapi/openapi3"
+	"github.com/iancoleman/strcase"
+	"golang.org/x/tools/go/packages"
+)
+
+type APIServiceFollowFunc struct {
+	Name           string
+	FirstParamName string
+	FirstParamType string
+	ReturnList     string
+	Src            string
+}
+
+type Follower struct {
+	APIService    string
+	APIRequest    string
+	APIFollowFunc APIServiceFollowFunc
+}
+
+type Followers = []Follower
+
+type FollowersParams = struct {
+	Followers
+}
+
+const followFuncComment = `// Follows a collection link. This function is based on the ` + "`%s`" + ` function, with one key difference:
+// instead of using a fixed endpoint path, ` + "`localVarPath`" + ` is defined as the base URL + the ` + "`link`" + ` provided in the arguments.`
+
+func AddLinkFollowersToParams(d *Data) error {
+	return AddValuesToParams(d, func(swagger *openapi3.T) (interface{}, error) { return getLinkFollowersParams(swagger) }, "linkfollowers.go")
+}
+
+func getLinkFollowersParams(swagger *openapi3.T) (params FollowersParams, err error) {
+	ids, idsErr := getOperationIDs(swagger)
+	if idsErr != nil {
+		err = idsErr
+		return
+	}
+
+	funcNameMap := getFuncNameReplacementMap(ids)
+
+	followers, followersErr := getLinkFollowers(funcNameMap)
+	if followersErr != nil {
+		err = followersErr
+		return
+	}
+
+	params = FollowersParams{
+		followers,
+	}
+
+	return
+}
+
+func generateExecuteFnName(in string) string {
+	// 'ToCamel()' mimics openapi-generator's 'camelize(sanitizeName(operationId))' logic for transforming OperationID.
+	// The "Execute" suffix is hardcoded in the template,
+	// see https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/go/api.mustache
+	return strcase.ToCamel(in) + "Execute"
+}
+
+func generateFollowFuncName(in string) string {
+	return "Follow" + strings.TrimSuffix(strings.TrimPrefix(strcase.ToCamel(in), "List"), "Execute") + "Link"
+}
```

Comment:
Don't do string concatination as it it is easy to make mistakes.

Use `fmt.Sprintf()` and use the formatters instead. This follows the convention we use everywhere else